libs/ui-toolkit/src/components/input/input.tsx

```diff
@@ -61,13 +106,70 @@ export const inputStyle = ({
       }
     : style;
 
+const getAffixElement = ({
+  prependElement,
+  prependIconName,
+  prependIconDescription,
+  appendElement,
+  appendIconName,
+  appendIconDescription,
+}: Pick<InputProps, keyof AffixProps>) => {
+  const position = prependIconName || prependElement ? 'pre' : 'post';
+
+  const className = classNames(
+    ['fill-black-60 dark:fill-white-60', 'absolute', 'z-10'],
+    {
+      'left-8': position === 'pre',
+      'right-8': position === 'post',
+    }
+  );
+
+  const element = prependElement || appendElement;
+  const iconName = prependIconName || appendIconName;
+  const iconDescription = prependIconDescription || appendIconDescription;
+
+  if (element) {
+    return <div className={className}>{element}</div>;
+  }
+
+  if (iconName) {
+    return (
+      <Icon
+        name={iconName}
+        className={className}
+        size={16}
+        aria-label={iconDescription}
+        aria-hidden={!iconDescription}
+      />
+    );
+  }
+
+  return null;
+};
+
 export const Input = forwardRef<HTMLInputElement, InputProps>(
-  ({ prependIconName, appendIconName, className, hasError, ...props }, ref) => {
+  (
+    {
+      prependIconName,
+      prependIconDescription,
+      appendIconName,
+      appendIconDescription,
+      prependElement,
+      appendElement,
+      className,
+      hasError,
+      ...props
+    },
+    ref
+  ) => {
     className = `h-28 ${className}`;
```

Comment:
suggestion (non blocking): use classnames func here for the hasPrepended, hasAppended logic?

Reply:
totally makes sense, changed it. also changed the className to be a variable, i think it's generally a bad practice to overwrite react props.